### PR TITLE
fix: parseHeaders should take Record<string, ...>

### PR DESCRIPTION
**Pull request by ronag**
## Status

- [ ] I have read and agreed to the [Developer's Certificate of Origin][cert]
- [ ] Tested
- [ ] Benchmarked (**optional**)
- [ ] Documented
- [ ] Review ready
- [ ] In review
- [ ] Merge ready

[cert]: https://github.com/nodejs/undici/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Review by mcollina — COMMENTED** [review]
Can you add a test? What is this fixing?

File: lib/core/util.js [hunk 1]
```diff
@@ -278,23 +278,50 @@ function bufferToLowerCasedHeaderName (value) {
...
+      if (val) {
+        if (typeof val === 'string') {
+          val = [val]
+          obj[key] = val
+        }
+        assert(!Array.isArray(val2))
+        val.push(val2.toString())
+      } else {
+        // TODO (fix): assert type...
+        obj[key] = Array.isArray(val2) ? val2.map(x => x.toString()) : val2.toString()
```
Comment (Uzlopak) [inline lib/core/util.js:306]: This works? What if val2 is null? Is this possible?

**Comment by ronag** [comment]
> Can you add a test? What is this fixing?

It's fixing so that the function does what the types say. Since we export this function it's important that it's correct.

**Comment by ronag** [comment]
Added in nxt-undici for now. If someone wants to makes tests feel free to take over.

**Pull request closed without merging**